test/java/src/com/ibm/streamsx/rest/test/StreamingAnalyticsConnectionTest.java

```diff
@@ -0,0 +1,117 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2017
+ */
+
+package com.ibm.streamsx.rest.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeNotNull;
+import static org.junit.Assume.assumeTrue;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.ibm.streamsx.rest.Instance;
+import com.ibm.streamsx.rest.RESTException;
+import com.ibm.streamsx.rest.StreamingAnalyticsConnection;
+
+public class StreamingAnalyticsConnectionTest extends StreamsConnectionTest {
+
+    // Tests that access StreamingAnalyticsConnection need to be overridden in
+    // this class
+    StreamingAnalyticsConnection connection = null;
+
+    @Override
+    public void setupConnection() {
+        try {
+            if (connection == null) {
+                String serviceName = System.getenv("STREAMING_ANALYTICS_SERVICE_NAME");
+                String vcapServices = System.getenv("VCAP_SERVICES");
+
+                // if we don't have serviceName or vcapServices, skip the test
+                assumeNotNull(serviceName, vcapServices);
+
+                testType = "STREAMING_ANALYTICS_SERVICE";
+                connection = StreamingAnalyticsConnection.createInstance(vcapServices, serviceName);
+            }
+        } catch (IOException e) {
+            fail("This shouldn't cause an exception");
```

Comment:
The test will fail due to an exception, but there will be not details on why it failed ...

Hiding exceptions is rarely a good idea in java - been burnt too many times on that